### DISCUSSION TITLE
Mouse right-click button delay?

### DISCUSSION THREAD

**Discussion by rafikiphoto** [sezanzeb/input-remapper#740] [Q&A] [answered]
I have an uncontrollable twitch in my right hand middle finger which constantly causes me to click the right mouse button, needing frequent cancellation of the right-click menu. Is there a way to add a delay to the right click button so that the right click button needs a longer press to activate? I manage this in Windows with X-Mouse Button Control and I'm wondering if I can use input-remapper to do the same in Linux Mint Cinnamon 21.1

**Comment by hxka** [comment]
Yes, use something like this: "if_tap(,hold_keys(BTN_RIGHT),300)"
300 specifies a delay in ms you'd need to hold a button for.

**Comment by rafikiphoto** [comment]
Thank you. Unfortunately I see no change. However, I'm sure it's my inexperience using the input remapper app so I'll continue trying.

------- Original Message -------

**Comment by rafikiphoto** [comment]
This is how I have it set up: see attached screenshot.

**Comment by hxka** [comment, marked as answer]
Github doesn't support e-mail attachments, so I can't see what you're doing. Here's what I did, it works:

![Screenshot_20230715_112305](https://github.com/sezanzeb/input-remapper/assets/6567590/e7e4eb60-79c6-4207-a8bb-302c06253ea1)

Note the comma before hold_keys, the first parameter to if_tap needs to be empty.

**Comment by rafikiphoto** [comment]
Thank you so much. I had made some silly mistakes but with a bit of trial and error it is now working perfectly as expected. I set the delay to 600 milliseconds, the same delay set in Windows X-Mouse and it works perfectly. It make such a difference to my life. Thanks again.
